internal/server/server.go

```diff
@@ -238,6 +242,10 @@ func (s *AsyncServer) eventLoop(ctx context.Context) error {
 
 // acceptConnection accepts a new client connection and subscribes to read events on the connection.
 func (s *AsyncServer) acceptConnection() error {
+	if len(s.connectedClients) > s.maxClients {
```

Comment:
Ideally it's not a good practice have same number of maxClients value as the backlog queue max number which is being set during `listen()`. Shouldn't be a blocker for this PR, but we should revisit this later.